src/lexer.py

```diff
@@ -290,6 +339,136 @@ def word_symbolizer(source: str, in_idx: int, error_generator: ErrorGenerator) -
+
+def tokenize_poetic_assignment(source: str, line: int, line_start: int, start_idx: int) -> Tuple[int, datatypes.Token]:
+    """
+    Assuming we've started a poetic type literal or poetic number assignment, attempt to tokenize
+    the remainder of the line as a type literal or poetic number. It is an error for a
+    poetic type literal assignment to include words after the type literal.
+
+    :param source:                   String of the source file
+    :param line:                     Index of the line in the source file, used to construct location
+    :param line_start:               Index of the first character in this line, used to construct location
+    :param start_index:              The index in the source file to start tokenizing from
+    :return:                         Tuple containing the next index to tokenize from and the token produced
+    :raises datatypes.LexerError:    On text that looked like it was a number or string but failed to parse correctly,
+                                     as well as when extra text follows the poetic type literal.
+    """
+    # At this point we don't know if this is a poetic type literal assignment or a
+    # poetic number assignment. To figure it out, we try to tokenize the first word
+    # as a type literal. If it fails, it must be a poetic number assignment.
+    success, result_idx, result_token = try_type_literal_assignment(source, line, line_start, start_idx)
+    if success:
+        assert result_token is not None
+        # This if check is needed to convert the Optional[Token] to just Token
+        if result_token is not None:
+            return result_idx, result_token
+
+    source_length = len(source)
+    idx = start_idx
+    current_word_length = 0
+    full_number_as_text = ""
+    added_decimal_point = False
+
+    while idx < source_length and source[idx] != "\n":
+        char = source[idx]
+        # If it's a space, add the current word length (if any) and reset
+        if char.isspace():
+            if current_word_length:
+                full_number_as_text += str(current_word_length)
+                current_word_length = 0
+        # If its a letter, increment the current word length (mod 10)
+        elif char.isalpha():
+            current_word_length = (current_word_length + 1) % 10
+        # If its a decimal point and we haven't added one, add a decimal
+        # point and start a new word
+        elif char == "." and not added_decimal_point:
+            if current_word_length:
+                full_number_as_text += str(current_word_length)
+                current_word_length = 0
+            full_number_as_text += "."
+            added_decimal_point = True
+        idx += 1
+    if current_word_length:
+        full_number_as_text += str(current_word_length)
+    if not full_number_as_text:
+        full_number_as_text = "0"
+    location = get_srcloc(line, line_start, start_idx, idx)
+    value = datatypes.Number(full_number_as_text)
```

Comment:
The data for a number is a number not a string

Reply:
It should be getting converted to a number. Tested and it seemed to work. Also I followed how you did it in parse_number.